frontend/src/components/AuthRoute/AuthRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Navigate } from 'react-router-dom';
+import { useGetProfile } from 'api/profile';
+import { ROLE } from 'constants/menuRoles';
+
+interface IAuthRouteProps {
+  children: React.ReactNode;
+  roles?: Array<ROLE>;
+}
+
+const AuthRoute: React.FC<IAuthRouteProps> = ({ children, roles }) => {
+  const { data } = useGetProfile();
+  const userHasRequiredRole = roles?.includes(data?.role as ROLE);
+  return userHasRequiredRole ? <>{children}</> : <Navigate to="/not-found" />;
+};
```

Comment:
Maybe will be better
```suggestion
const RoleRoute = ({ path, element, roles }) => {
  const { data: userProfileResponse } = useGetProfile();

  const isAllowed = roles.includes(userProfileResponse?.role as ROLE);

  return isAllowed ? <Route path={path} element={element}> : <Navigate to={PATHS.notFound} />;
};

export default RoleRoute;
```